lms/services/course.py

```diff
@@ -151,6 +151,55 @@ def get_organization_courses(
             .all()
         )
 
+    def _deduplicated_course_assigments_query(self, courses: list[Course]):
```

Comment:
Extract this to a helper method. Only change is that it takes a list of courses now.